Managers/FrameMan.cpp

```diff
@@ -1,2651 +1,1391 @@
+
+			// Only draw this stuff for actual human players
+			if (whichScreen < g_ActivityMan.GetActivity()->GetHumanCount()) {
+				// Team of current screen's player
+				//int team = g_ActivityMan.GetActivity()->GetTeamOfPlayer(g_ActivityMan.GetActivity()->PlayerOfScreen(whichScreen));
+
+				yTextPos += 12;
+
+				// Message
+				if (!m_ScreenText[whichScreen].empty()) {
+					if (IsInMultiplayerMode()) {
+						if (m_TextCentered[whichScreen]) { yTextPos = (GetPlayerFrameBufferHeight(whichScreen) / 2) - 52; }
+
+						int occOffsetX = g_SceneMan.GetScreenOcclusion(whichScreen).m_X;
+
+						// Draw blinking effect, but not of the text message itself, but some characters around it (so it's easier to read the message)
+						if (m_TextBlinking[whichScreen] && m_TextBlinkTimer.AlternateReal(m_TextBlinking[whichScreen])) {
+							GetLargeFont()->DrawAligned(&pPlayerGUIBitmap, (GetPlayerFrameBufferWidth(whichScreen) + occOffsetX) / 2, yTextPos, (">>> " + m_ScreenText[whichScreen] + " <<<").c_str(), GUIFont::Centre);
+						} else {
+							GetLargeFont()->DrawAligned(&pPlayerGUIBitmap, (GetPlayerFrameBufferWidth(whichScreen) + occOffsetX) / 2, yTextPos, m_ScreenText[whichScreen].c_str(), GUIFont::Centre);
+						}
+					} else {
+						if (m_TextCentered[whichScreen]) { yTextPos = (GetPlayerScreenHeight() / 2) - 52; }
+
+						int occOffsetX = g_SceneMan.GetScreenOcclusion(whichScreen).m_X;
+
+						// If there's really no room to offset the text into, then don't
+						if (GetPlayerScreenWidth() <= GetResX() / 2)
+							occOffsetX = 0;
+						// Draw blinking effect, but not of the text message itself, but some characters around it (so it's easier to read the message)
+						if (m_TextBlinking[whichScreen] && m_TextBlinkTimer.AlternateReal(m_TextBlinking[whichScreen])) {
+							GetLargeFont()->DrawAligned(&pPlayerGUIBitmap, (GetPlayerScreenWidth() + occOffsetX) / 2, yTextPos, (">>> " + m_ScreenText[whichScreen] + " <<<").c_str(), GUIFont::Centre);
+						} else {
+							GetLargeFont()->DrawAligned(&pPlayerGUIBitmap, (GetPlayerScreenWidth() + occOffsetX) / 2, yTextPos, m_ScreenText[whichScreen].c_str(), GUIFont::Centre);
+						}
+					}
+					yTextPos += 12;
+				}
+				g_PerformanceMan.Draw(pPlayerGUIBitmap);
+
+				// Draw info text when in MOID or material layer draw mode
+				switch (g_SceneMan.GetLayerDrawMode()) {
+					case g_LayerTerrainMatter:
+						GetSmallFont()->DrawAligned(&pPlayerGUIBitmap, GetPlayerScreenWidth() / 2, GetPlayerScreenHeight() - 12, "Viewing terrain material layer\nHit Ctrl+M to cycle modes", GUIFont::Centre, GUIFont::Bottom);
+						break;
+					case g_LayerMOID:
+						GetSmallFont()->DrawAligned(&pPlayerGUIBitmap, GetPlayerScreenWidth() / 2, GetPlayerScreenHeight() - 12, "Viewing MovableObject ID layer\nHit Ctrl+M to cycle modes", GUIFont::Centre, GUIFont::Bottom);
+						break;
+					default:
+						break;
+				}
 
+				// If superfluous screen (as in a three-player match), make the fourth the Observer one
```

Comment:
Bad tabbing again. Tbh I'd move this down to be right above line 440 (the one that does the drawing) since it makes more sense there.

Also, kill the commented code below. Alternatively, and probably preferably, set yTextPos to 0 from the beginning and kill the yTextPos += 12 stuff in the first half of the if-else, since it's fine if the observer text is shifted down 12 pixels.